### DISCUSSION TITLE
Quasar Q-Btn and Tailwindcss on Safari doesn't display properly when using :to

### DISCUSSION THREAD

**Discussion by maggie44** [tailwindlabs/tailwindcss#5609] [Help] [answered]
**Platform:**
Quasar Version:
@quasar/app Version: 2.0.4

**What version of Tailwind CSS are you using?**

    "tailwindcss": "^2.2.15"

**What version of Node.js are you using?**

v14.17.2

**What browser are you using?**

Safari (Desktop and Mobile)
Safari: Version 14.1.1 (16611.2.7.1.4)

**What operating system are you using?**

macOS

**Reproduction URL**

I have tried to replicate this in JSFiddle but as it is Safari browser specific I haven't been able to. Instead, I have attached a demo project. (`yarn install` and `quasar dev` to start)

[safari-issue 2.zip](https://github.com/tailwindlabs/tailwindcss/files/7231786/safari-issue.2.zip)

**Describe your issue**
When using `to` or `:to` on a q-btn (in Quasar) the style changes when viewing through Safari (iPhone and Desktop Safari) (Firefox works ok), and doesn't seem to respond to the style requests.

<img width="119" alt="Screenshot 2021-09-10 at 08 36 39" src="https://user-images.githubusercontent.com/64841595/132880132-634a6b20-6aa0-4605-a7cc-a6b7381d760f.png">

I raise this with the Quasar team and they suggested it is an issue with the Tailwinds styling being applied: https://github.com/quasarframework/quasar/issues/10636

**To Reproduce**

It is a Quasar project created using the Quasar CLI + a tailwinds setup as described here (https://dev.to/eltoritoelvis/add-tailwind-css-to-your-quasar-project-in-5-minutes-or-less-3n4o). Then try the below two examples:

```
    <q-btn
      icon="folder_open"
      outline
      dense
      flat
      to="/"
    />
    <q-btn
      icon="folder_open"
      outline
      dense
      flat
    />
```

**Expected behavior**
Both these buttons should look the same like they do on Firefox (tested with Developer Edition):
<img width="208" alt="Screenshot 2021-09-10 at 08 54 59" src="https://user-images.githubusercontent.com/64841595/132882367-c2936500-462b-44ae-98ab-1a351d2748e7.png">

**Comment by adamwathan** [comment, marked as answer]
Hey! This looks like a Quasar issue — they are generating invalid HTML when you add the `to` attribute:

![image](https://user-images.githubusercontent.com/4323180/134820761-5b6a4f4d-6627-4f8f-bcf1-ed6b270860f6.png)

Notice the `type="button"` on the `a` tag, anchor tags do not support that attribute. Tailwind sets anything with `type="button"` to be styled with a`-webkit-appearance: button` by default. Quasar should not be including that attribute on links, only on buttons.

One workaround for your project is to add this CSS rule to your `base` layer:

```css
@layer base {
  [type="button"] {
    -webkit-appearance: unset;
  }
}
```

...but this may have unintended consequences for real buttons.

Another is to add `appearance: none` to anything with the `q-btn` class:

```css
.q-btn {
  appearance: none;
}
```

Ultimately though the best long term fix is to get Quasar to patch the issue where they are adding the type attribute incorrectly to `<a>` tags 👍🏻

**Reply by pdanpdan** [reply]
Found the problem, thank you.

**Reply by maggie44** [reply]
Thanks for investigating. 

Adding here again the link to the tracking issue over on the Quasar repo for others to find: https://github.com/quasarframework/quasar/issues/10636
